Lesson_3/FIOandEMAI.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace Lesson_3
+{
+    internal class FIOandEMAI
+    {
+        public List<string> Loader(FileInfo file)
+        {
+            List<string> listS = new List<string>();
+            using (StreamReader sr = new StreamReader(file.FullName, Encoding.Default))
+            {
+                while (!sr.EndOfStream)
+                {
+                    listS.Add(sr.ReadLine());
+                }
+            }
+            return listS;
+        }
+        public List<Information> Desereales(List<string> listS)
+        {
+            List<Information> linfo = new List<Information>();
+            foreach (string s in listS)
+            {
+                Information inform = new Information(s.Split('&').First(), s.Split('&').Last());
+                linfo.Add(inform);
+            }
+            return linfo;
+        }
+        public void CreateFileEmail(List<string> list)
+        {
+            var arr = list.ToArray();
+            for (int i = 0; i < list.Count(); i++)
+            {
+                SearchMail(ref arr[i]);
+            }
+            string dir = Directory.GetCurrentDirectory();
+            dir = Path.Combine(dir, "Data");
+            if (!Directory.Exists(dir))
+            {
+                Directory.CreateDirectory(Path.Combine(dir, "Data"));
+            }
```

Comment:
```suggestion
            Directory.CreateDirectory(Path.Combine(dir, "Data"));
```
Можно просто сразу создавать директорию. Если её не было, то она будет создана, а если была, то ничего не произойдёт.

За `Path.Combine` отдельный 👍